### PR TITLE
Create entities and config

### PR DESCRIPTION
**Pull request by Digicat13**
Created entities 
Added configurations for entities
Created initial migration

### CODE REVIEW

**Comment by Digicat13** [comment]
Added abstract BaseEntity<T> with property "Id" of type T
Other Entities inherit from abstract BaseEntity
Changed List<Entity> to IEnumerable<Entity>
Removed "virtual" keyword

File: QuizApp.DAL/Configuration/TestConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using QuizApp.DAL.Entities;
+
+namespace QuizApp.DAL.Configuration
+{
+	public class TestConfiguration: IEntityTypeConfiguration<Test>
```
Comment (olenarehushleobit) [inline QuizApp.DAL/Configuration/TestConfiguration.cs:10]: Before `:` should be space.
Please format all your files using `Cntrl+K+D`

File: QuizApp.DAL/Configuration/UserConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            builder.HasKey(e => e.Id);
+            builder.Property(e => e.Id).ValueGeneratedOnAdd();
+            builder.Property(e => e.Username)
+                .HasMaxLength(100)
+                .IsRequired();
+            builder.Property(e => e.Password)
+                .HasMaxLength(50)
+                .IsRequired();
+        }
+
```
Comment (olenarehushleobit) [inline QuizApp.DAL/Configuration/UserConfiguration.cs:23]: Extra l. Please remove

File: QuizApp.DAL/Configuration/TestAnswerConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+			builder.Property(e => e.AnswerText)
+	 			.IsRequired();
+			builder.Property(e => e.IsCorrect)
+				.IsRequired();
+
+			builder.HasOne(ta => ta.TestQuestion)
+				.WithMany(tq => tq.TestAnswers)
+				.HasForeignKey(ta => ta.TestQuestionId)
+				.OnDelete(DeleteBehavior.NoAction);
+
```
Comment (olenarehushleobit) [inline QuizApp.DAL/Configuration/TestAnswerConfiguration.cs:27]: extra line

File: QuizApp.DAL/Entities/TestingUrl.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	{
+		public Guid TestId { set; get; }
+		public string IntervieweeName { set; get; }
+		public DateTime? AllowedStartDate { set; get; }
+		public DateTime? AllowedEndDate { set; get; }
+		public int NumberOfRuns { set; get; }
+
+		public Test Test { set; get; }
+		public IEnumerable<TestingResult> TestingResults { get; set; }
+
```
Comment (olenarehushleobit) [inline QuizApp.DAL/Entities/TestingUrl.cs:18]: extra space

File: QuizApp.DAL/QuizAppContext.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			modelBuilder.ApplyConfiguration(new TestConfiguration());
+			modelBuilder.ApplyConfiguration(new TestAnswerConfiguration());
+			modelBuilder.ApplyConfiguration(new TestingResultConfiguration());
+			modelBuilder.ApplyConfiguration(new TestingResultAnswerConfiguration());
+			modelBuilder.ApplyConfiguration(new TestingUrlConfiguration());
+			modelBuilder.ApplyConfiguration(new TestQuestionConfiguration());
+			modelBuilder.ApplyConfiguration(new UserConfiguration());
+		}
+
```
Comment (olenarehushleobit) [inline QuizApp.DAL/QuizAppContext.cs:34]: extra space

File: QuizApi/QuizApi.sln [hunk 6]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (olenarehushleobit) [inline QuizApi/QuizApi.sln:1]: test comment

**Review by olenarehushleobit — COMMENTED** [review]
test comment

**Comment by Digicat13** [comment]
Removed extra lines
Initializing IEnumerables with empty Lists
Changed { set; get; } in some properties with { get; set; }
Formatted files

**Review by olenarehushleobit — APPROVED** [review]

**Pull request merged**